### PR TITLE
Migration from MSEng to DevDiv - release/1.0.0

### PR DESCRIPTION
**Pull request by ravimeda**
These changes migrate build of core-setup release 1.0.0 from the current MSEng to DevDiv.  Changes include:

1. Build definitions checked in under buildpipeline folder
2. Signing project is in build_projects/dotnet-host-build/sign.proj
3. Updated Agent Pool to DotNetCore-Test.
4. Fixed a yum package conflict.

@gkhanna79 please provide feedback.

### CODE REVIEW

File: build_projects/dotnet-host-build/sign.proj [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="SignCoreHost" DependsOnTargets="GetSignCoreHostFiles">
+    <CallTarget Targets="SignFiles" />
+  </Target>
+
+  <Target Name="GetSignCoreHostFiles">
+    <ItemGroup>
+      <FilesToSign Include="$(OutDir)corehost/**/hostfxr.dll">
```
Comment (gkhanna79) [inline build_projects/dotnet-host-build/sign.proj]: Thanks - please add a comment here that dotnet.exe is not intentionally being signed.

**Comment by gkhanna79** [comment]
@dagood @wtgodbe @ramarag PTAL as well.

CC @leecow 

@ravimeda We should not merge this until the branch reopens.

**Comment by wtgodbe** [comment]
LGTM

File: buildpipeline/Core-Setup-Signing-Windows-x64.json [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+    "url": "https://github.com/dotnet/core-setup.git",
+    "defaultBranch": "release/1.0.0",
+    "clean": "true",
+    "checkoutSubmodules": false
+  },
+  "quality": "definition",
+  "queue": {
+    "pool": {
+      "id": 83,
+      "name": "DotNetCore-Test"
```
Comment (naamunds) [inline buildpipeline/Core-Setup-Signing-Windows-x64.json]: Is this going to change to `DotNet-Build` once that pool has Windows Server 2016 machines?  Ditto for the other two Windows build definitions.

Comment (ravimeda) [inline buildpipeline/Core-Setup-Signing-Windows-x64.json]: I'm not sure of the future plan here.  But Dan L and DDIT are aware of this usage, and will advise if Pool configuration changes.

**Comment by ravimeda** [comment]
https://ci.dot.net/job/dotnet_core-setup/job/master/job/debug_opensuse42.1_x64_prtest/698/
https://ci.dot.net/job/dotnet_core-setup/job/master/job/release_ubuntu_arm_prtest/55/
https://ci.dot.net/job/dotnet_core-setup/job/master/job/release_ubuntu16.10_x64_prtest/695/
@dotnet-bot test this please.

**Comment by ravimeda** [comment]
`OpenSUSE42.1:x64:Debug`, `Ubuntu:arm:Release` and `Ubuntu16.10:x64:Release` are not in the list of platforms to cover as per https://github.com/dotnet/core-setup/blob/release/1.0.0/netci.groovy#L12

I think that explains the failures. @mmitche please could you confirm.

**Comment by gkhanna79** [comment]
Those are not supported for 1.0.0. Please see CI for https://github.com/dotnet/core-setup/pull/1669 to ensure the right set of platforms are in place for CI.

**Comment by mmitche** [comment]
There are 3 different errors there.  1) can't find the script to run 2) can't download something 3) failed to initialize coreclr.  Those may be consistent with not being supported.

File: build_projects/dotnet-host-build/sign.proj (hunk 1 above)
Comment (ravimeda) [inline build_projects/dotnet-host-build/sign.proj]: Comment added through commit `8fed9ff`.

**Comment by ravimeda** [comment]
Thanks @mmitche @gkhanna79

**Comment by ravimeda** [comment]
Commit is `5716c1f` bring the new Docker image for RHEL to address a locale issue.  Refer to #1682

**Comment by ravimeda** [comment]
@MichaelSimons please could you review `5716c1f`

**Comment by dagood** [comment]
The 503 GitHub errors in
https://ci.dot.net/job/dotnet_core-setup/job/release_1.0.0/job/release_windows_nt_x64_prtest/203/
https://ci.dot.net/job/dotnet_core-setup/job/release_1.0.0/job/debug_windows_nt_x86_prtest/203/
have happened to me before when pulling an empty file from GitHub (https://github.com/dotnet/buildtools/issues/1196), but that isn't the case here. (https://raw.githubusercontent.com/dotnet/cli/rel/1.0.0-preview2/scripts/obtain/dotnet-install.ps1) I guess that bug has other ways to hit it now--I actually just repro'd it on my machine in my browser, although my first request succeeded.

```
Invoke-WebRequest : 
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
 "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html>
  <head>
    <title>503 first byte timeout</title>
  </head>
  <body>
    <h1>Error 503 first byte timeout</h1>
    <p>first byte timeout</p>
    <h3>Guru Mediation:</h3>
    <p>Details: cache-sjc3138-SJC 1489078076 902614549</p>
    <hr>
    <p>Varnish cache server</p>
  </body>
</html>
At D:\j\workspace\release_windo---d8901c9b\build_projects\dotnet-host-build\build.ps1:82 char:1
```

@dotnet-bot test Windows_NT x64 Release Build
@dotnet-bot test Windows_NT x86 Debug Build

**Comment by gkhanna79** [comment]
Same concern about updating RHEL7 image as explained at https://github.com/dotnet/core-setup/pull/1603#issuecomment-285416957

**Comment by ravimeda** [comment]
RHEL concern is addressed. Hence merging.

**Pull request merged**